### PR TITLE
Adds culling conditionals post

### PR DESCRIPTION
**Pull request by samjonester**

### CODE REVIEW

**Comment by samjonester** [comment]
Waiting to merge until Brittany's post has had some time up front.

File: app/posts/2017-12-13-culling-conditionals.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+1. How each of these specific actions affect the larger system since none of them have names.
+
+Just thinking about all the things I need to think about leaves me exhausted. This is another reason why conditionals spread. After getting a grasp on all that information, I just don't have any more room to plan a refactor!
+
+## Abstractions to Keep the Population at Bay
+
+Just like those carp, conditionals should be carefully culled. Unfortunately, Slack doesn't have integrations for [electric barriers or poison pills](http://www.chicagotribune.com/news/local/breaking/ct-lake-michigan-asian-carp-met-20160822-story.html). There are a few patterns that I regularly use instead of conditionals. Before blindly adopting any of these techniques, ask if the code can be written [without a conditional](http://michaelfeathers.typepad.com/michael_feathers_blog/2013/11/unconditional-programming.html).
+
+These 3 techniques build on [All the Little Things](https://www.youtube.com/watch?v=8bZh5LMaSm://www.youtube.com/watch?v=8bZh5LMaSmE). They are ways to design components that have a singular focus, rather than a list of responsibilities like we see above.
```
Comment (rosston) [inline app/posts/2017-12-13-culling-conditionals.md]: Fixed link: https://www.youtube.com/watch?v=8bZh5LMaSmE

File: app/posts/2017-12-13-culling-conditionals.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+Each technique has a specific problem it solves. Each proposed alternative has a few important things in common.
+
+- The new abstraction has a single purpose. Choose and execute the appropriate path(s) through the application. That's it. Fake paths can be provided for testing the abstraction so details about a specific path aren't complicating the test.
+- Each path has the same 2 methods. One to test whether it should be executed, and one to execute the action. This allows each piece to be defined and tested in isolation. Also we gain valuable names describing each path, helping other developers understand it's purpose.
```
Comment (rosston) [inline app/posts/2017-12-13-culling-conditionals.md]: "it's" should be "its"

File: app/posts/2017-12-13-culling-conditionals.md (hunk 1 above)
Comment (rosston) [reply app/posts/2017-12-13-culling-conditionals.md]: Also, this line says "these 3 techniques", but there appear to be only 2 techniques listed.

**Pull request merged**